src/main.ts

```diff
@@ -280,10 +288,31 @@ export default class ReadwisePlugin extends Plugin {
     }
     // close the ZipReader
     await zipReader.close();
+    await this.acknowledgeSyncCompleted(buttonContext)

```

Comment:
What happens if we have some error in the javascript above (or eg the user quits Obsidian) while syncing, and this never gets called?

The client will just not sync for the next hour, right? (until lock on our server expires?)

Reply:
@tadeoos ok, but is this necessary though? would it not be possible for the server to clear the lock, so this client try again without waiting a whole hour?

Reply:
yeah, I changed the behavior to just continue if we have a lock from the same client already